src/protocols/bdx/BDXMessageUtils.cpp

```diff
@@ -0,0 +1,408 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Implements utility methods for working with some complex BDX messages.
+ */
+
+#include <protocols/bdx/BDXMessageUtils.h>
+
+#include <support/BufBound.h>
+#include <support/BufferReader.h>
+#include <support/CodeUtils.h>
+
+#define VERSION_MASK 0x0F
+#define SENDER_DRIVE_MASK 0x10
+#define RECEIVER_DRIVE_MASK 0x20
+#define ASYNC_MASK 0x40
+#define CONTROL_MODE_MASK 0xF0
+
+#define DEFLEN_MASK 0x01
+#define START_OFFSET_MASK 0x02
+#define WIDERANGE_MASK 0x10
+
+using namespace chip;
+using namespace chip::BDX;
+using namespace chip::Encoding::LittleEndian;
+
+CHIP_ERROR TransferInit::Pack(System::PacketBuffer & aBuffer)
+{
+    CHIP_ERROR err   = CHIP_NO_ERROR;
+    uint8_t rangeCtl = 0;
+    uint8_t ptcByte  = 0;
+
+    BufBound bbuf(aBuffer.Start(), aBuffer.AvailableDataLength());
+
+    ptcByte |= mSupportedVersions & VERSION_MASK;
+    if (mSupportsSenderDrive)
```

Comment:
We could use BitFlags in the main structure and then this bool to byte conversion would not be needed.

bools take 1 byte each where as bitflags would compact them. Minor size benefit.

Reply:
I think BitFlags may improve readability as well. I'll make the change